source/develop/release-management/features/storage/gluster-multiple-bricks-per-storage.md

```diff
@@ -0,0 +1,74 @@
+---
+title: Support for multiple GlusterFS bricks on a block device(s)
+category: feature
+authors: dchaplyg
+feature_name: Support for multiple GlusterFS bricks on a block device(s)
+feature_modules: engine/vdsm
+feature_status: Planning
+---
+
+# Support for multiple GlusterFS bricks on a block device(s)
+
+# Summary
+
+This feature introduces ability to create multiple GlusterFS bricks on the same block device(s).
+
+In case of shared block device, engine should create a LVM volume group on top of it and use LVM logical volumes as a brick backend devices. That approach will allow management of the local storage alongside with support for thin provisioning for the GlusterFS and SSD caching.
+
+## Owner
+
+*   Name: [Denis Chaplygin](https://github.com/akashihi)
+*   Email: <dchaplyg@redhat.com>
+
+## Current status
+
+*   Target Release: 4.3
+*   Status: Planning
+*   Last updated: December 19, 2017
+
+# Detailed description
+
+Current oVirt's GlusterFS brick/volume management implementation implies that bricks should occupy the whole block device, thus limiting possible number of bricks to the number of physically available block devices. That approach could be quite limiting, especially in case of small deployments or hyperconverged environments, when you have just one or two relatively big drives for you VM payloads.
+
+With this change we will introduce LVM management support into oVirt, thus allowing users to manage their local storage using oVirt UI and use LVM logical volumes as a block devices for GlusterFS bricks. Due to LVM ability it will also bring LVM specific features, such as thin provisioning and SSD caching.
+
+# Implementation details
+
+## Engine
+
+* New LVM specific entities should be added to represent managed LVM data.
+* Database access layer needs to be extended to store per-host managed LVM data
+* Engine should support querying LVM status and configuration on hosts, manage physical volumes, volume groups and logical volumes. This can be implemented using engine's Ansible integration, to not to pollute vdsm with that code.
```

Comment:
It's not clear how Ansible will return the information to Engine...

Reply:
Well, we will gather facts on the ansible side and return them as json string. On the engine side we will parse that string to a Map and work with the map data.